smart-contracts/contracts/cl-vault/src/helpers.rs

```diff
@@ -132,6 +129,53 @@ pub fn get_deposit_amounts_for_liquidity_needed(
     ))
 }
 
+// this math is straight from the readme
+pub fn get_single_sided_deposit_0_to_1_swap_amount(
+    storage: &dyn Storage,
+    querier: &QuerierWrapper,
+    token0_balance: Uint128,
+    lower_tick: i64,
+    upper_tick: i64,
+) -> Result<Uint128, ContractError> {
+    let spot_price = Decimal256::from(get_spot_price(storage, querier)?);
+    let lower_price = tick_to_price(lower_tick)?;
+    let upper_price = tick_to_price(upper_tick)?;
+    let pool_metadata_constant: Uint128 = spot_price
+        .checked_mul(lower_price.sqrt())?
+        .checked_mul(upper_price.sqrt())?
+        .to_uint_floor() // todo: this is big, so should be safe, right?
+        .try_into()?;
+
+    let swap_amount = token0_balance.checked_multiply_ratio(
+        pool_metadata_constant,
+        pool_metadata_constant.checked_add(Uint128::one())?,
+    )?;
+
+    Ok(swap_amount)
+}
+
+pub fn get_single_sided_deposit_1_to_0_swap_amount(
+    storage: &dyn Storage,
+    querier: &QuerierWrapper,
+    token1_balance: Uint128,
+    lower_tick: i64,
+    upper_tick: i64,
+) -> Result<Uint128, ContractError> {
+    let spot_price = Decimal256::from(get_spot_price(storage, querier)?);
+    let lower_price = tick_to_price(lower_tick)?;
+    let upper_price = tick_to_price(upper_tick)?;
+    let pool_metadata_constant: Uint128 = spot_price
```

Comment:
is the spot price the same unit as lower price and upper price?

Reply:
@nseguias can you verify if this is/is not the case ?

Reply:
Spot price is represented as String of Decimal (18 decimal points). Lower & Upper price is represented as Decimal256.

```
osmosisd q poolmanager spot-price 1 uosmo ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2 --node https://rpc.osmosis.zone:443 -o json | jq
{
  "spot_price": "0.052326079000000000"
}
```

```
pub struct SpotPriceResponse {
    /// String of the Dec. Ex) 10.203uatom
    ...
}
```

```
pub fn tick_to_price(tick_index: i64) -> Result<Decimal256, ContractError> { ... }
```